### ISSUE TITLE
Fully implement #style

### ISSUE THREAD

**Issue by PragTob** [shoes/shoes4#623]
We should fully implement the `#style` method.

`#style` is a dual purpose accessor, where `#style` returns the hash of the styles currently applied to the object in their original form (Meaning it also returns things like `height: "100%"` not the calculated valued). 

On the other hand `#style(a_style_hash)` sets values. I do believe we already have a `Common::Style` module that doesn't work with everything though (see #622 for instance).

One "problem" here is that a couple of the styles reside in another object, namely the `Dimensions` object has all the margins/widths/heights/lefts/rights whatnot.

In short, `style` is an interface to restyle **EVERYTHING** and get the styles of everything as well.
